### PR TITLE
TINY-9860: Use non-bogus <br> to pad last table cell

### PR DESCRIPTION
**Pull request by ArvinJ-H**
Related Ticket: 
https://ephocks.atlassian.net/browse/TINY-9860

Description of Changes:
remove `data-mce-bogue`for last table cell

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] <s>Docs ticket created (if applicable)</s>

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts [hunk 1]
```diff
@@ -190,7 +190,12 @@ const moveSelectionToMarker = (editor: Editor, marker: HTMLElement | null): void
       rng = nextRng;
       dom.remove(parentBlock);
     } else {
-      dom.add(parentBlock, dom.create('br', { 'data-mce-bogus': '1' }));
+      // If parentBlock is a table cell, add a br without 'data-mce-bogus' attribute.
+      if (isTableCell(parentBlock)) {
+        dom.add(parentBlock, dom.create('br'));
+      } else {
+        dom.add(parentBlock, dom.create('br', { 'data-mce-bogus': '1' }));
+      }
```
Comment (mkzhx) [inline modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts:198]:
```suggestion
      dom.add(parentBlock, dom.create('br', isTableCell(parentBlock) ? {} : { 'data-mce-bogus': '1' }));
```

File: modules/tinymce/src/plugins/table/test/ts/browser/TableRemoveTrailingBrTest.ts
Comment (mkzhx) [inline modules/tinymce/src/plugins/table/test/ts/browser/TableRemoveTrailingBrTest.ts:1]: Can you add a test for `remove_trailing_br: true` and asserting that there is no `<br>` in any table cell?

**Review by mkzhx — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md [hunk 2]
```diff
@@ -39,7 +39,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The `caption`, `address` and `dt` elements no longer incorrectly allow non-inline child elements when the editor schema is set to _HTML 4_. #TINY-9768
 - SVG icons for back and foreground colors now use `class` instead of `id` to identify SVG elements that should change color. #TINY-9844
 - Anchor tag elements — `<a>` — no longer incorrectly allow non-inline child elements when the editor schema is set to _HTML 4_. #TINY-9805
-- Help dialog was restored to `medium` width for better readability. #TINY-9947
+- Help dialog was restored to `medium` width for better readability #TINY-9947
+- Stopped adding `data-mce-bogus` attribute to `br` in empty table cells #TINY-9860.
```
Comment (mkzhx) [inline modules/tinymce/CHANGELOG.md:43]: The changelog entry should be what was changed from the customer's perspective. In this case we fixed the bug of `remove_trailing_brs: true` not removing the trailing `<br>` of an empty last cell of a table. Write an entry to that sense and add it under `Fixed`. 
```suggestion
```

**Review by mkzhx — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md [hunk 3]
```diff
@@ -39,7 +39,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The `caption`, `address` and `dt` elements no longer incorrectly allow non-inline child elements when the editor schema is set to _HTML 4_. #TINY-9768
 - SVG icons for back and foreground colors now use `class` instead of `id` to identify SVG elements that should change color. #TINY-9844
 - Anchor tag elements — `<a>` — no longer incorrectly allow non-inline child elements when the editor schema is set to _HTML 4_. #TINY-9805
-- Help dialog was restored to `medium` width for better readability. #TINY-9947
+- Help dialog was restored to `medium` width for better readability #TINY-9947
```
Comment (tiny-ben-tran) [inline modules/tinymce/CHANGELOG.md:42]:
```suggestion
- Help dialog was restored to `medium` width for better readability. #TINY-9947
```

It will undo the change, which should not be in here

File: modules/tinymce/src/plugins/table/test/ts/browser/TableRemoveTrailingBrTest.ts [hunk 4]
```diff
@@ -40,6 +40,21 @@ describe('browser.tinymce.plugins.table.TableRemoveTrailingBrTest', () => {
...
+      TinyAssertions.assertContentPresence(editor, { br: 4 });
+    });
+
+    it('TINY-9860: nested table, last table cell is br, not removing br', async () => {
+      const editor = hookWithoutTrailingBr.editor();
+      editor.setContent('');
+      await TableTestUtils.pInsertTableViaGrid(editor, 2, 2);
+      await TableTestUtils.pInsertTableViaGrid(editor, 2, 2);
+      TinyAssertions.assertContentPresence(editor, { br: 7 });
+    });
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/table/test/ts/browser/TableRemoveTrailingBrTest.ts:57]: I don't really see a difference between the new tests that were added and the existing tests in this file. I think we can rework this test file to save some duplication

File: modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts [hunk 5]
```diff
@@ -190,7 +190,8 @@ const moveSelectionToMarker = (editor: Editor, marker: HTMLElement | null): void
       rng = nextRng;
       dom.remove(parentBlock);
     } else {
-      dom.add(parentBlock, dom.create('br', { 'data-mce-bogus': '1' }));
+      // If parentBlock is a table cell, add a br without 'data-mce-bogus' attribute.
+      dom.add(parentBlock, dom.create('br', isTableCell(parentBlock) ? {} : { 'data-mce-bogus': '1' }));
```
Comment (ltrouton) [inline modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts:194]: The `isTableCell` check is done in the above `if` statement as well. Might be worth having a variable like `const blockIsTableCell = isTableCell(parentBlock);` that can be used in both places

File: modules/tinymce/src/plugins/table/test/ts/browser/TableRemoveTrailingBrTest.ts [hunk 6]
```diff
@@ -5,41 +5,111 @@ import Editor from 'tinymce/core/api/Editor';
...
     base_url: '/project/tinymce/js/tinymce'
   };
-
-  context('remove_trailing_br with nested table', () => {
-    const hookWithTrailingBr = TinyHooks.bddSetup<Editor>({ ...baseSettings, remove_trailing_br: true }, [ Plugin ], true);
-    const hookWithoutTrailingBr = TinyHooks.bddSetup<Editor>({ ...baseSettings, remove_trailing_br: false }, [ Plugin ], true);
-
-    it('TINY-3909: single table, not removing br', async () => {
-      const editor = hookWithoutTrailingBr.editor();
+  context('remove_trailing_brs: true', () => {
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/table/test/ts/browser/TableRemoveTrailingBrTest.ts:13]:
```suggestion

  context('remove_trailing_brs: true', () => {
```

File: modules/tinymce/src/plugins/table/test/ts/browser/TableRemoveTrailingBrTest.ts
Comment (ltrouton) [inline modules/tinymce/src/plugins/table/test/ts/browser/TableRemoveTrailingBrTest.ts:1]: There are a lot of missing newlines between the `context` and `it` blocks for the new file changes.

File: modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts [hunk 7]
```diff
@@ -181,16 +181,19 @@ const moveSelectionToMarker = (editor: Editor, marker: HTMLElement | null): void
...
     rng.setStart(parentBlock, 0);
     rng.setEnd(parentBlock, 0);
 
-    if (!isTableCell(parentBlock) && !isPartOfFragment(parentBlock) && (nextRng = findNextCaretRng(rng))) {
+    if (!isCell && !isPartOfFragment(parentBlock) && (nextRng = findNextCaretRng(rng))) {
       rng = nextRng;
       dom.remove(parentBlock);
     } else {
-      dom.add(parentBlock, dom.create('br', { 'data-mce-bogus': '1' }));
+      // If parentBlock is a table cell, add a br without 'data-mce-bogus' attribute.
```
Comment (mkzhx) [inline modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts:195]: Add the Jira ticket number to this comment. Instead of stating the obvious, explain that it is to keep it consistent with the <br> in other empty table cells and for `remove_trailing_brs` to function correctly.

**Review by mkzhx — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts (hunk 7 above)
Comment (ArvinJ-H) [reply modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts:195]:
```suggestion
      // TINY-9860: If parentBlock is a table cell, add a br without 'data-mce-bogus' attribute.
```

**Review by tiny-ben-tran — APPROVED** [review]

**Review by TheSpyder — APPROVED** [review]
